source/geometry_model/spherical_shell.cc

```diff
@@ -216,6 +216,23 @@ namespace aspect
                                               R1,
                                               0,
                                               true);
+
+          // Tell p4est about the periodicity of the mesh.
+          std::vector<GridTools::PeriodicFacePair<typename parallel::distributed::Triangulation<dim>::cell_iterator> >
+          periodicity_vector;
```

Comment:
this looks wrong. Why do you have two different variables here and use them differently below? Delete this line.